src/libslic3r/MeshBoolean.cpp

```diff
@@ -200,12 +200,12 @@ indexed_triangle_set cgal_to_indexed_triangle_set(const _Mesh &cgalmesh)
     const auto &vertices = cgalmesh.vertices();
     int vsize = int(vertices.size());
 
-    for (auto &vi : vertices) {
+    for (const auto &vi : vertices) {
```

Comment:
Hi,

Could you elaborate a bit more on the issue you're addressing? I'm not entirely clear on how changing a vertex_index/face_index reference to a constant impacts this particular case.
I didn't see build issues either

Reply:
Hi. I was trying to build on Fedora using system dependencies. CGAL 5.6 contains [this commit](https://github.com/CGAL/cgal/commit/d95d834af10559a58ac928e6fb50619d4fe352ab), which causes a compilation errors:
```
OrcaSlicer/src/libslic3r/MeshBoolean.cpp:203:5: error: cannot bind non-const lvalue reference of type ‘CGAL::SM_Vertex_index&’ to an rvalue of type ‘boost::iterators::detail::iterator_facade_base<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epick> >::Index_iterator<CGAL::SM_Vertex_index>, CGAL::SM_Vertex_index, std::random_access_iterator_tag, CGAL::SM_Vertex_index, long int, false, false>::reference’ {aka ‘CGAL::SM_Vertex_index’}
  203 |     for (auto &vi : vertices) {
      |     ^~~
OrcaSlicer/src/libslic3r/MeshBoolean.cpp:208:5: error: cannot bind non-const lvalue reference of type ‘CGAL::SM_Face_index&’ to an rvalue of type ‘boost::iterators::detail::iterator_facade_base<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epick> >::Index_iterator<CGAL::SM_Face_index>, CGAL::SM_Face_index, std::random_access_iterator_tag, CGAL::SM_Face_index, long int, false, false>::reference’ {aka ‘CGAL::SM_Face_index’}
  208 |     for (auto &face : faces) {
      |     ^~~
```
If it helps, PrusaSlicer made the same change:
https://github.com/prusa3d/PrusaSlicer/commit/4aa7366fafb7a3f91cc68d4a808f5195a4aa3dcf

Reply:
Thanks for the details. 
Now I see the issue; there was a risk of modifying a temporary variable. 
It looks like the newer version of GCC is much more intelligent 👍